packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/components/FormApiTokenContainer.tsx

```diff
@@ -1,27 +1,67 @@
-import React from 'react';
+import * as React from 'react';
 
 import { Box, Flex, Grid, GridItem, Typography } from '@strapi/design-system';
-import PropTypes from 'prop-types';
+import { FormikErrors } from 'formik';
 import { useIntl } from 'react-intl';
 
-import LifeSpanInput from '../../../../../components/Tokens/LifeSpanInput';
-import TokenDescription from '../../../../../components/Tokens/TokenDescription';
-import TokenName from '../../../../../components/Tokens/TokenName';
-import TokenTypeSelect from '../../../../../components/Tokens/TokenTypeSelect';
+// @ts-expect-error not converted yet
+import LifeSpanInput from '../../../../components/Tokens/LifeSpanInput';
+// @ts-expect-error not converted yet
+import TokenDescription from '../../../../components/Tokens/TokenDescription';
+// @ts-expect-error not converted yet
+import TokenName from '../../../../components/Tokens/TokenName';
+// @ts-expect-error not converted yet
+import TokenTypeSelect from '../../../../components/Tokens/TokenTypeSelect';
 
-const FormApiTokenContainer = ({
-  errors,
+type TokenType = 'full-access' | 'read-only' | 'custom';
+
+interface FormApiTokenContainerProps {
+  errors?: FormikErrors<{
+    name?: string;
+    description?: string;
+    lifespan?: string;
+    type?: string;
```

Comment:
Does it make sense the do one FormValues type you pass around so they're better in sync? You use them in multiple places is all 👀